### ISSUE TITLE
tags layer can't work well with go lang

### ISSUE THREAD

**Issue by imbeder** [wsdjeg/SpaceVim#2044]
## Expected behavior, english is recommend
after install tags layer, with go lang files, we can: jump to definition, find references, etc

## Environment Information

- OS: mac
- vim version: 8.1.250
- neovim version: -
- SpaceVim version: 0.9.0-dev
- SpaceVim status: 

```
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   config/init.vim

no changes added to commit (use "git add" and/or "git commit -a")
```

## The reproduce ways from Vim starting (Required!)
1, enable tags layer, install global with pygments and universal ctags support, install universal ctags
2, open go project
3, find references or jump to definition, not work

## Output of the `:SPDebugInfo!`

<details><summary> SpaceVim debug information </summary>

### SpaceVim options :

```toml
  auto_disable_touchpad = 1
  autocomplete_method = 'neocomplete'
  autocomplete_parens = 1
  background = 'dark'
  buffer_index_type = 4
  checkinstall = 1
  colorscheme = 'gruvbox'
  colorscheme_bg = 'dark'
  colorscheme_default = 'desert'
  custom_color_palette = []
  custom_plugins = []
  debug_level = 1
  default_indent = 4
  disabled_plugins = []
  enable_ale = 0
  enable_cursorcolumn = 0
  enable_cursorline = 1
  enable_debug = 0
  enable_googlesuggest = 0
  enable_guicolors = 1
  enable_insert_leader = 1
  enable_javacomplete2_py = 0
  enable_key_frequency = 0
  enable_language_specific_leader = 1
  enable_neocomplcache = 0
  enable_neomake = 1
  enable_os_fileformat_icon = 0
  enable_powerline_fonts = 1
  enable_statusline_display_mode = 0
  enable_statusline_mode = 0
  enable_tabline_filetype_icon = 0
  enable_vimfiler_filetypeicon = 0
  enable_vimfiler_gitstatus = 0
  enable_vimfiler_welcome = 1
  enable_ycm = 0
  error_symbol = '✖'
  expand_tab = 1
  filemanager = 'nerdtree'
  filetype_icons = {}
  force_global_config = 0
  gitcommit_issue_icon = ''
  gitcommit_pr_icon = ''
  github_username = ''
  guifont = ''
  hiddenfileinfo = 1
  hosts_url = 'https://raw.githubusercontent.com/racaljk/hosts/master/hosts'
  info_symbol = 'ⓘ'
  keep_server_alive = 1
  language = ''
  leader_guide_default_group_name = ''
  leader_guide_flatten = 1
  leader_guide_hspace = 5
  leader_guide_max_size = 0
  leader_guide_position = 'botright'
  leader_guide_run_map_on_popup = 1
  leader_guide_sort_horizontal = 0
  leader_guide_submode_mappings = {'<C-C>': 'win_close'}
  leader_guide_vertical = 0
  lint_on_save = 1
  lint_on_the_fly = 0
  max_column = 120
  plugin_bundle_dir = '/Users/imbeder/.cache/vimfiles/'
  plugin_manager = 'dein'
  plugin_manager_processes = 16
  plugin_name = 'gtags.vim'
  project_rooter_automatically = 1
  project_rooter_patterns = ['.git/', '_darcs/', '.hg/', '.bzr/', '.svn/', '.clang']
  realtime_leader_guide = 1
  relativenumber = 1
  search_tools = ['rg', 'ag', 'pt', 'ack', 'grep']
  sidebar_width = 30
  simple_mode = 0
  smartcloseignoreft = ['tagbar', 'vimfiler', 'SpaceVimRunner', 'SpaceVimREPL', 'SpaceVimQuickFix', 'HelpDescribe', 'VebuggerShell', 'VebuggerTerminal', 'SpaceVimTabsManager']
  smartcloseignorewin = ['__Tagbar__', 'vimfiler:default']
  snippet_engine = 'neosnippet'
  src_root = 'E:\sources\'
  statusline_inactive_separator = 'arrow'
  statusline_left_sections = ['winnr', 'filename', 'major mode', 'syntax checking', 'minor mode lighters']
  statusline_right_sections = ['fileformat', 'cursorpos', 'percentage']
  statusline_separator = 'arrow'
  statusline_unicode_symbols = 1
  terminal_cursor_shape = 2
  version = '0.9.0-dev'
  vim_help_language = 'en'
  vimcompatible = 0
  warning_symbol = '⚠'
  wildignore = '*/tmp/*,*.so,*.swp,*.zip,*.class,tags,*.jpg,*.ttf,*.TTF,*.png,*/target/*,.git,.svn,.hg,.DS_Store,*.svg'
  windows_index_type = 0
  windows_leader = 's'
  windows_smartclose = 'q'
```

### SpaceVim layers :

```toml
[[layers]]
  name="autocomplete"
  auto-completion-return-key-behavior='complete'
  auto-completion-tab-key-behavior='smart'
[[layers]]
  name="checkers"
[[layers]]
  name="format"
[[layers]]
  name="edit"
[[layers]]
  name="ui"
[[layers]]
  name="core"
[[layers]]
  name="core#banner"
[[layers]]
  name="core#statusline"
[[layers]]
  name="core#tabline"
[[layers]]
  name="shell"
  default_position='top'
  default_height=30
[[layers]]
  name="lang#go"
[[layers]]
  name="lang#c"
[[layers]]
  name="lang#html"
[[layers]]
  name="lang#lua"
[[layers]]
  name="tags"
[[layers]]
  name="-l"
```

### SpaceVim Health checking :

SpaceVim clipboard support check report:
Checking +clipboard:
      SUCCEED!
SpaceVim environment check report:
Current progpath: vim(vim)
version: 801
OS: mac
[shell, shellcmdflag, shellslash]: ['/bin/zsh', '-c', 0]
SpaceVim lua support check report:
Checking +lua:
      SUCCEED!
SpaceVim python support check report:
Checking +python3:
      SUCCEED!
Checking +python:
      Failed : to support +python, Please install vim-gik, or build from sources.

### SpaceVim runtime log :

```log
[ SpaceVim ] : logger file  does not exists, only log for current process will be shown!
[ SpaceVim ] [16:45:50] [ Info ] Can not find project local config, start to loadding global config
```
</details>

## Screenshots

If you have any screenshots for this issue please upload here. BTW you can use https://asciinema.org/ for recording video  in terminal.

**Comment by imbeder** [comment]
global (GNU GLOBAL) 6.5.6
Copyright (c) 1996-2016 Tama Communications Corporation
License GPLv3+: GNU GPL version 3 or later <http://www.gnu.org/licenses/gpl.html>
This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Universal Ctags 0.0.0(a0206b0), Copyright (C) 2015 Universal Ctags Team
Universal Ctags is derived from Exuberant Ctags.
Exuberant Ctags 5.8, Copyright (C) 1996-2009 Darren Hiebert
  Compiled: Aug 14 2018, 11:13:38
  URL: https://ctags.io/
  Optional compiled features: +wildcards, +regex, +iconv, +option-directory, +xpath, +case-insensitive-filenames

GTAGSLABEL=pygments
GTAGSCONF=/usr/local/share/gtags/gtags.conf

Above is some version info and related env. I have tried the newest global, and your documents command to install global with brew, but that command can't support universal ctags, so I build brew rb file with universal ctags support.

Tags layer can work with c/c++ projects, but not go projects. In go project, after generate GTAGS,  I can find the tags in the GTAGS file, but 'SPC+m+g+d' can't work.

One more strange thing: with https://github.com/derekparker/delve project, even I don't generate GTAGS file, the tag layer can work very well, including jump to definition, find references... I tried to find the difference between this go project and other go project, find that if delve project's .git file is removed, then tags layer can't work, even I delete all the file in .git.... 
unbelievable...
